File: Array/ChefandNotebooks.cpp
#include<iostream>
using namespace std;

int main()
{
    int test,x,y,cash,n,page,price,flag;
    //x= page needed
    //y= page alreasy has
    // cash =available balance
    //n = number of notebooks shopkeeper showd

    cin>>test;
    while(test--)
    {
        flag=0;
        cin>>x>>y>>cash>>n;

        for(int j=0;j<n;j++)
        {
            cin>>page>>price;
            if(page>=(x-y) && price<= cash)
            {
                flag=1;
            }
        }
        if(flag==0)
            cout<<"UnluckyChef"<<endl;
        else
            cout<<"LuckyChef"<<endl;
    }

}

File: Array/ChefandNotebooks.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Dec 16 22:44:24 2020

@author: gourob
"""

test =int(input())

for i in range(test):
    flag=0
    x,y,cash,n=map(int,input().split())
    for j in range(n):
        page, price = map(int,input().split())
        if(page>=(x-y) and price <=cash):
            flag=1
            
    if flag==0:
        print("UnluckyChef")
    else:
        print("LuckyChef")
